### ISSUE TITLE
can I only see the ascii ? 

### ISSUE THREAD

**Issue by jiuxiaxixi** [snarlistic/jpnevulator#3]
I want use it to capture the data (only ascii )from boards in background, and I use the Hex output would be difficult for me，could you please tell me how to disable the HEX output?Thanks a lot!!

**Comment by snarlistic** [comment]
Hi jiuxiaxixi,

On Sun, Jan 08, 2017 at 07:07:42AM -0800, jiuxiaxixi wrote:
> could you please tell me how to disable the HEX output?

I'm not yet sure if I feel this is a nice feature I want to
support, but you can do this quiet easily yourself. Simply
comment out both lines 485 and 490 in the file jpnevulator.c and
compile the source. This should produce an executable that will
disable the hexadecimal output.

In the mean time I will think if I like to support this feature.
Please forgive me that I'm not yet enthousiastic about it.


-- 
$ cat ~/.signature
Freddy Spierenburg <freddy@snarl.nl>         http://freddy.snarl.nl/
GnuPG fingerprint: C71E F201 840C 8998 4D3F F675 B507 BF7D 9218 273E
$ # Please read http://www.ietf.org/rfc/rfc2015.txt before complain!

**Comment by snarlistic** [comment]
I've thought about it for a fairly long time and I'm sorry, I don't really like this. Please forgive me. :)

**Comment by mariobaldini** [comment]
@jiuxiaxixi I tried a workaround this this command:
`sudo jpnevulator --tty /dev/ttyUSB0 --read | xxd -a -r -p`
it will read the hex output and convert to ascii.

But unfortunatelly jpevulator (or something else) is not acting totally passive and when I run it in parallel with a ongoing connection, I can read the output (only in ascii, for instance), but loosing some packets, etc...

Ie. when 'screen' opens a device attached to /dev/ttyUSB0, data flow works fine. 

If I use the above command in a separated terminal, I can get some data in the jpnevulator... and also some still in the 'screen', but both corrupted. When I close jpevulator, screen flow resume to the ok state. 

Maybe the above command has some bug. I tried MANY other approaches, using sed, awk, etc ... to remove \n.. prepend 0x to force hex convertion, etc... 

but in the end since the original 'screen' to device normal reading gets corrupted if jpenvulator is opened in parallel eventually put me on hold with this approach.

**Comment by snarlistic** [comment]
Hi Mario,

Please see the [manual page](https://jpnevulator.snarl.nl/src/current/manual.html) and look for the --ascii option. There is no need to use an external program like xdd to convert the hexadecimal output to ascii.

Please also see the first question answered in the [FAQ](https://jpnevulator.snarl.nl/src/current/FAQ). It will help you understand why you can't make two separate processes read from the same device using GNU/Linux.

However, I see no reason why you can't process the output of jpnevulator as input within another piece of software. Hence my reasoning why I refrained from supported a no hexadecimal output option. First I think its close to useless and second I think one can easily write a filter that drops it if it's really important to drop that representation of the data. Again, it's easy to both display hexadecimal and ascii data.

Sincerely,
Freddy
